cmd/cigarToBed/cigarToBed.go

```diff
@@ -0,0 +1,198 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/align"
+	"github.com/vertgenlab/gonomics/bed" //TODO: raven added this line for cigarToBed
+	"github.com/vertgenlab/gonomics/fasta"
+	"github.com/vertgenlab/gonomics/fileio"
+	"github.com/vertgenlab/gonomics/genomeGraph"
+	"log"
+	"os"      //raven added this line for MSA Fasta output
+	"strings" //raven added this line for CountSeqIdx
+)
+
+//raven did not put this helper function into the globalAlignment function because it is used twice within the globalAlignment function
+//raven wrote this block to count sequences based on the Read function in gonomics/fasta/fasta.go
+//raven changed the input variable from filename string to inputFile EasyReader, so that the file is only opened 1 time for 2 purposes: faDone and CountSeqIdx
+func CountSeqIdx(inputFile *fileio.EasyReader) int {
+	var line string
+	var seqIdx int = 1 //I know in Read seqIdx is int64 and starts with -1, but I am using it differently here. EasyReader comes in having read the first fasta, so seqIdx starts with 1
+	var doneReading bool = false
+	defer inputFile.Close()
+	for line, doneReading = fileio.EasyNextRealLine(inputFile); !doneReading; line, doneReading = fileio.EasyNextRealLine(inputFile) {
+		if strings.HasPrefix(line, ">") {
+			seqIdx++
+		}
+	}
+	return seqIdx
+}
+
+//raven did not put this helper function into the globalAlignment function because it is used in the globalAlignment function
+//faONe is target, faTwo is query
+func cigarToGraph(target fasta.Fasta, query fasta.Fasta, aln []align.Cigar) *genomeGraph.GenomeGraph {
+	answer := genomeGraph.EmptyGraph()
+	//use targetEnd and queryEnd to track position number for each fasta sequence as we move along.
+	var targetEnd, queryEnd int = 0, 0
+	var curr *genomeGraph.Node
+
+	//Creating the first node. This is done independent of all other number because this node has no 'previous' nodes.  All following code leverages the cigar output (second number printed in the Op position of the struct {}) to determine if the alignment returned a 0:match, 1:insertion, or 2:deletion. All indels are relative to the target sequence.
```

Comment:
Now that I've seen Dan do it, you could break up this big comment line into multiple lines that fit better on the screen (instead of having to scroll right). But 100% a stylistic thing.

Reply:
Yes I remember seeing it too! I'll do that moving forward.